src/pages/tabbed-dashboard/App.js

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useRef, useCallback, useEffect } from 'react';
+import styles from './submitForm.module.css';
+import TokenButton from '../../navigation/TokenButton/TokenButton';
+import MenuButton from '../../navigation/MenuButton/MenuButton';
+import * as Callbridge from '@iotum/callbridge-js';
+import { useSelector } from 'react-redux';
+
+const App = () => {
+  const [isYourAppVisible, setIsYourAppVisible] = useState(true);
+  const [unreadMessages, setUnreadMessages] = useState(0);
+  const [isWidgetInitialized, setIsWidgetInitialized] = useState(false);
+  const [chatWidgetReady, setChatWidgetReady] = useState(false);
+
+  const containerRef = useRef(null);
+  const chatWidgetRef = useRef(null);
+  const widgetRef = useRef(null);
+
+  const credentials = useSelector(state => state.credentials);
+
+  const unloadWidgets = () => {
+    if (widgetRef.current) {
+      widgetRef.current.toggle(false);
+    }
+    if (chatWidgetRef.current) {
+      chatWidgetRef.current.toggle(false);
+    }
+  };
+
+  const renderWidget = useCallback(() => {
+    widgetRef.current = new Callbridge.Dashboard({
+      domain: credentials.domain,
+      sso: {
+        token: credentials.token,
+        hostId: credentials.hostId,
+      },
+      container: containerRef.current,
+    });
+
+    widgetRef.current.toggle(false);
+
+    widgetRef.current.on('dashboard.READY', () => {
+      chatWidgetRef.current = new Callbridge.Dashboard({
+        domain: 'iotum.callbridge.rocks',
+        container: containerRef.current,
+      }, "Team");
+
+      chatWidgetRef.current.toggle(false);
+
+      chatWidgetRef.current.on('dashboard.UNREAD_MESSAGES', (data) => {
+        const totalUnread = Object.values(data.rooms).reduce((total, count) => total + count, 0);
+        setUnreadMessages(totalUnread);
+      });
+
+      chatWidgetRef.current.on('dashboard.READY', () => {
+        setChatWidgetReady(true);
+        setIsWidgetInitialized(true);
+      });
+    });
+  }, [credentials]);
+
+  useEffect(() => {
+    if (isWidgetInitialized) {
+      chatWidgetRef.current.on('dashboard.NAVIGATE', (data) => {
+        const room = data.pathname !== '/' ? data.pathname : "an unspecified room";
+        console.log(`The chat widget navigated to this room: ${room}`);
+      });
+    }
+  }, [isWidgetInitialized]);
+
+  useEffect(() => {
+    renderWidget();
+  }, [credentials, renderWidget]);
```

Comment:
1. no need for `credentials` as dependency. since `renderWidget` already has it.
2. this app is missing widget.unload during unmount

generally speaking, to make life cycle management easier to read/understand/track, you could use this pattern:

```js
  useEffect(() => {
    widgetRef.current = new Callbridge.Dashboard({...});
    ...
    ...

    return () => {
      widgetRef.current.unload();
      widgetRef.current = null;
    };
  }, [credentials]);
```

Reply:
2. at least seems to now be satisfied, resolving